src/Symfony/Component/Intl/Tests/LanguagesTest.php

```diff
@@ -952,40 +1775,36 @@ public function testAlpha3CodeExists()
      */
     public function testGetAlpha3Name($displayLocale)
     {
-        $names = Languages::getNames($displayLocale);
+        $names = Languages::getAlpha3Names($displayLocale);
 
-        foreach ($names as $alpha2 => $name) {
-            $alpha3 = self::$alpha2ToAlpha3[$alpha2] ?? false;
-            if ($alpha3) {
-                $this->assertSame($name, Languages::getAlpha3Name($alpha3, $displayLocale));
-            }
+        foreach ($names as $language => $name) {
+            $this->assertSame($name, Languages::getAlpha3Name($language, $displayLocale));
         }
     }
 
-    public function testGetAlpha3NameWithInvalidCountryCode()
+    public function testGetAlpha3NameWithInvalidLanguageCode()
     {
         $this->expectException(MissingResourceException::class);
 
-        Languages::getAlpha3Name('ZZZ');
+        Languages::getAlpha3Name('zzz');
     }
 
     /**
      * @dataProvider provideLocales
      */
     public function testGetAlpha3Names($displayLocale)
     {
-        $names = Languages::getAlpha3Names($displayLocale);
+        $languages = array_keys(Languages::getAlpha3Names($displayLocale));
 
-        $alpha3Codes = array_keys($names);
-        sort($alpha3Codes);
-        $this->assertSame(array_values(self::$alpha2ToAlpha3), $alpha3Codes);
+        sort($languages);
+
+        $this->assertNotEmpty($languages);
+        $this->assertEmpty(array_diff($languages, self::$alpha3Codes));
 
-        $alpha2Names = Languages::getNames($displayLocale);
-        foreach ($alpha2Names as $alpha2Code => $name) {
-            if (!isset(self::$alpha2ToAlpha3[$alpha2Code])) {
-                unset($alpha2Names[$alpha2Code]);
+        foreach ($languages as $alpha3Code) {
+            if (isset(self::$alpha3ToAlpha2[$alpha3Code])) {
+                $this->assertSame(Languages::getName(self::$alpha3ToAlpha2[$alpha3Code], $displayLocale), Languages::getAlpha3Name($alpha3Code, $displayLocale));
```

Comment:
This tests `Languages::getAlpha3Name` not the values from `Languages::getAlpha3Names`

If you want this test, it should be moved to testGetAlpha3Name

Reply:
got ya, see 3ad5d3e1d8554d62e5f9292f290b0d839a4c9212

Reply:
Here you should test the actual result from `Languages::getAlpha3Names` and not throw it away at line 1797